gc/base/SweepPoolManagerAddressOrderedListBase.cpp

```diff
@@ -236,10 +245,18 @@ MM_SweepPoolManagerAddressOrderedListBase::connectChunk(MM_EnvironmentBase *env,
 						previousConnectChunk->trailingFreeCandidate);
 
 				connectChunkPostProcess(chunk, sweepState, (MM_HeapLinkedFreeHeader *)previousConnectChunk->trailingFreeCandidate, previousFreeEntry);
+
+				if (memoryPool->doesNeedClearCardForConnectFreeEntry()) {
+					memoryPool->cleanCardsForFreeEntry(env, (void*)previousFreeEntry, ((uint8_t*)previousFreeEntry + previousFreeEntrySize), false);
+				}
+
 				/* only for SPMSAOL */
 				previousPreviousFreeEntry = previousFreeEntry;
 				previousFreeEntry = (MM_HeapLinkedFreeHeader *)previousConnectChunk->trailingFreeCandidate;
 				previousFreeEntrySize = previousConnectChunk->trailingFreeCandidateSize;
+				if ((_extensions->isVLHGC()) && (env->_cycleState->_noCompactionAfterSweep)) {
```

Comment:
could this go up a bit, so it goes into VLHGC variant of connectChunkPostProcess ?